portality/api/v2/data_objects/application.py

```diff
@@ -0,0 +1,379 @@
+                        "oa_statement": {"coerce": "unicode"},
+                        "journal": {"coerce": "unicode"},
+                        "aims_scope": {"coerce": "unicode"},
+                        "author_instructions": {"coerce": "unicode"}
+
+                    }
+                },
+                "subject": {
+                    "fields": {
+                        "code": {"coerce": "unicode"},
+                        "scheme": {"coerce": "unicode"},
+                        "term": {"coerce": "unicode"}
+                    }
+                },
+                "waiver": {
+                    "fields": {
+                        "has_waiver": {"coerce": "unicode"},
+                        "url": {"coerce": "unicode"}
+                    }
+                }
+            }
+        }
+    }
+}
+
+INCOMING_APPLICATION_REQUIREMENTS = {
+    "required" : ["admin", "bibjson"],
+
+    "structs": {
+        "admin" : {
+            "required" : ["contact"],
+            "structs" : {
+                "contact": {
+                    "required" : ["name", "email"]
+                }
+            }
+        },
+
+        "bibjson": {
+            "required": [
+                "apc",
+                "copyright",
+                "deposit_policy",
+                "editorial",
+                "keywords",
+                "language",
+                "license",
+                "ref",
+                "pid_scheme",
+                "plagiarism",
+                "preservation",
+                "publication_time_weeks",
+                "publisher",
+                "ref",
+                "other_charges",
+                "waiver",
+                "title"
+            ],
+            "structs": {
+                "apc": {
+                    "required" : ["has_apc"]
+                },
+                "copyright": {
+                    "required" : ["url"]
+                },
+                "deposit_policy": {
+                    "required" : ["has_policy"]
+                },
+                "editorial": {
+                    "required" : ["review_process", "review_url"]
+                },
+                "pid_scheme": {
+                    "required" : ["has_pid_scheme"]
+                },
+                "plagiarism": {
+                    "required": ["detection","url"]
+                },
+                "preservation": {
+                    "required": ["has_preservation"]
+                },
+                "publisher": {
+                    "required": ["name"]
+                },
+                "ref": {
+                    "required" : ["journal"]
+                },
+                "other_charges": {
+                    "required": ["has_other_charges"]
+                },
+                "waiver": {
+                    "requred": ["has_waiver"]
+                },
+            }
+        }
+    }
+}
+
+
+class IncomingApplication(dataobj.DataObj, swagger.SwaggerSupport):
```

Comment:
I'll just make the comment here, but it applies everywhere that we use `DataObj`, is that we should use `SeamlessMixin` for the new version of the API (the v1 should continue to use `DataObj`).